### ISSUE TITLE
[ios] Unhandled Exception: PlatformException(unavailable, Only available in iOS 13.0 or newer, null, null)

### ISSUE THREAD

**Issue by aoatmon** [okadan/flutter-nfc-manager#52]
### given 
- flutter create nfc_ios
- add to ios\runner the file `Runner.entitlements` with 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.developer.nfc.readersession.formats</key>
	<array>
		<string>TAG</string>
	</array>
</dict>
</plist>
```
- add to plist 

```xml

	<!-- added for nfc START -->
	<key>NFCReaderUsageDescription</key>
	<string>Read NFC Tags</string>
	<!-- added for nfc STOP -->
	
```

- add code

```dart
import 'package:flutter/material.dart';
import 'package:nfc_manager/nfc_manager.dart';

void _justPrint() async {
  final instance = NfcManager.instance;
  if (await instance.isAvailable()) {
    instance.startSession(onDiscovered: (tag) async {
      for (final entry in tag.data.entries) {
        print('key ${entry.key}');
        print('value ${entry.value}');
      }
      print(tag.handle);
    });
  }
}

void main() => runApp(const MaterialApp(home: JustListen()));

class JustListen extends StatefulWidget {
  const JustListen({Key? key}) : super(key: key);

  @override
  _JustListenState createState() => _JustListenState();
}

class _JustListenState extends State<JustListen> {
  @override
  void initState() {
    _justPrint();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return const Material(child: Center(child: CircularProgressIndicator()));
  }
}

```

<details>
<summary>logs</summary>

```bash
[ +205 ms] [ 52%] CreatingStagingDirectory
[        ] [ 57%] ExtractingPackage
[        ] [ 60%] InspectingPackage
[  +57 ms] [ 60%] TakingInstallLock
[ +249 ms] [ 65%] PreflightingApplication
[        ] [ 67%] PatchingApplication
[ +164 ms] [ 67%] InstallingEmbeddedProfile
[  +23 ms] [ 70%] VerifyingApplication
[+1471 ms] [ 75%] CreatingContainer
[  +12 ms] [ 80%] InstallingApplication
[  +18 ms] [ 85%] PostflightingApplication
[   +6 ms] [ 90%] SandboxingApplication
[  +33 ms] [ 95%] GeneratingApplicationMap
[ +948 ms] [100%] Installed package build/ios/iphoneos/Runner.app
[ +165 ms] ------ Debug phase ------
[        ] Starting debug of 42f545c80499280e43186687e3c51ad278ab2985 (N61AP, iPhone 6 (GSM), iphoneos, arm64) a.k.a. 'xxxx iPhone' connected through USB...
[ +535 ms] [  0%] Looking up developer disk image
[  +30 ms] [ 95%] Developer disk image mounted successfully
[ +726 ms] [100%] Connecting to remote debug server
[        ] -------------------------
[  +34 ms] (lldb) command source -s 0 '/tmp/E059A27B-9BA1-41D5-8C6B-6DF7CAEA822A/fruitstrap-lldb-prep-cmds-42f545c80499280e43186687e3c51ad278ab2985'
[        ] Executing commands in '/tmp/E059A27B-9BA1-41D5-8C6B-6DF7CAEA822A/fruitstrap-lldb-prep-cmds-42f545c80499280e43186687e3c51ad278ab2985'.
[        ] (lldb)     platform select remote-ios --sysroot '/Users/xxx/Library/Developer/Xcode/iOS DeviceSupport/12.4.4 (16G140)/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/xxx/Library/Developer/Xcode/iOS DeviceSupport/12.4.4 (16G140)/Symbols"
[        ] (lldb)     target create "/Users/xxx/development/nfc_ios/build/ios/iphoneos/Runner.app"
[+3686 ms] Current executable set to '/Users/xxx/development/nfc_ios/build/ios/iphoneos/Runner.app' (arm64).
[        ] (lldb)     script fruitstrap_device_app="/private/var/containers/Bundle/Application/BC09B299-4916-4348-884A-388256129132/Runner.app"
[ +347 ms] (lldb)     script fruitstrap_connect_url="connect://127.0.0.1:58751"
[        ] (lldb)     script fruitstrap_output_path=""
[        ] (lldb)     script fruitstrap_error_path=""
[        ] (lldb)     target modules search-paths add /usr "/Users/xxx/Library/Developer/Xcode/iOS DeviceSupport/12.4.4 (16G140)/Symbols/usr" /System "/Users/xxx/Library/Developer/Xcode/iOS DeviceSupport/12.4.4
(16G140)/Symbols/System" "/private/var/containers/Bundle/Application/BC09B299-4916-4348-884A-388256129132" "/Users/xxx/development/nfc_ios/build/ios/iphoneos" "/var/containers/Bundle/Application/BC09B299-4916-4348-884A-388256129132"
"/Users/xxx/development/nfc_ios/build/ios/iphoneos" /Developer "/Users/xxx/Library/Developer/Xcode/iOS DeviceSupport/12.4.4 (16G140)/Symbols/Developer"
[  +12 ms] (lldb)     command script import "/tmp/E059A27B-9BA1-41D5-8C6B-6DF7CAEA822A/fruitstrap_42f545c80499280e43186687e3c51ad278ab2985.py"
[   +3 ms] (lldb)     command script add -f fruitstrap_42f545c80499280e43186687e3c51ad278ab2985.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_42f545c80499280e43186687e3c51ad278ab2985.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_42f545c80499280e43186687e3c51ad278ab2985.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_42f545c80499280e43186687e3c51ad278ab2985.safequit_command safequit
[        ] (lldb)     connect
[  +37 ms] (lldb)     run
[ +271 ms] success
[        ] (lldb)     safequit
[ +120 ms] Process 403 detached
[  +33 ms] Application launched on the device. Waiting for observatory port.
[+1259 ms] Observatory URL on device: http://127.0.0.1:54922/
[   +7 ms] Attempting to forward device port 54922 to host port 58761
[        ] executing: /Users/xxx/fvm/versions/stable/bin/cache/artifacts/usbmuxd/iproxy 58761:54922 --udid 42f545c80499280e43186687e3c51ad278ab2985
[+1014 ms] Forwarded port ForwardedPort HOST:58761 to DEVICE:54922
[        ] Forwarded host port 58761 to device port 54922 for Observatory
[   +3 ms] Installing and launching... (completed in 11.7s)
[   +2 ms] Caching compiled dill
[  +76 ms] Connecting to service protocol: http://127.0.0.1:58761/
[ +376 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:58761/.
[ +111 ms] DDS is listening at http://127.0.0.1:58764/NxGQy33a8q0=/.
[  +59 ms] Successfully connected to service protocol: http://127.0.0.1:58761/
[  +27 ms] DevFS: Creating new filesystem on the device (null)
[  +49 ms] DevFS: Created new filesystem on the device (file:///private/var/mobile/Containers/Data/Application/C8206186-F545-463E-906F-5D6E2A2052B0/tmp/nfc_ios3cHwNg/nfc_ios/)
[   +2 ms] Updating assets
[  +92 ms] Syncing files to device xxx iPhone...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:nfc_ios/main.dart 53d5d210-693c-4a5d-b854-49157bbed721
[        ] <- 53d5d210-693c-4a5d-b854-49157bbed721
[  +65 ms] Updating files.
[        ] DevFS: Sync finished
[   +2 ms] Syncing files to device xxx iPhone... (completed in 70ms)
[        ] Synced 0.0MB.
[   +1 ms] <- accept
[  +12 ms] Connected to _flutterView/0x14e037e20.
[   +3 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[        ] An Observatory debugger and profiler on xxx iPhone is available at: http://127.0.0.1:58764/NxGQy33a8q0=/
[ +540 ms] The Flutter DevTools debugger and profiler on xxx iPhone is available at: http://127.0.0.1:9101?uri=http%3A%2F%2F127.0.0.1%3A58764%2FNxGQy33a8q0%3D%2F
[ +264 ms] [VERBOSE-2:ui_dart_state.cc(199)] Unhandled Exception: PlatformException(unavailable, Only available in iOS 13.0 or newer, null, null)
[        ] #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:597:7)
[        ] #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:158:18)
[        ] <asynchronous suspension>
[        ] #2      _justPrint (package:nfc_ios/main.dart:6:7)
[        ] <asynchronous suspension>
[+18713 ms] DevFS: Deleting filesystem on the device (file:///private/var/mobile/Containers/Data/Application/C8206186-F545-463E-906F-5D6E2A2052B0/tmp/nfc_ios3cHwNg/nfc_ios/)
[  +30 ms] DevFS: Deleted filesystem on the device (file:///private/var/mobile/Containers/Data/Application/C8206186-F545-463E-906F-5D6E2A2052B0/tmp/nfc_ios3cHwNg/nfc_ios/)
[ +119 ms] Service protocol connection closed.
[        ] Application finished.
[   +5 ms] "flutter run" took 47,754ms.
[  +85 ms] ensureAnalyticsSent: 82ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

</details>

<details>
<summary>doctor</summary>

```console
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-x64, locale en-EE)
    • Flutter version 2.2.2 at /Users/xxx/fvm/versions/stable
    • Framework revision d79295af24 (5 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/xxx/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.57.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (3 available)
    • xxx’s iPhone (mobile) • xxx • ios            • iOS 12.4.4
    • macOS (desktop)             • macos                                    • darwin-x64     • macOS 11.4 20F71 darwin-x64
    • Chrome (web)                • chrome                                   • web-javascript • Google Chrome 91.0.4472.106
```

</details>

<details>
<summary>pubspec</summary>

```yaml
name: nfc_ios
description: A new Flutter project.
version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  nfc_manager: ^3.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true
```

</details>

### note

same code works perfectly on android

**Comment by okadan** [comment]
> Only available in iOS 13.0 or newer

Check your iOS version.

**Comment by aoatmon** [comment]
> > Only available in iOS 13.0 or newer
> 
> Check your iOS version.

@okadan 
i see... closing for now

> Note: This plugin depends on NFCTagReaderSession (requires iOS 13.0 or later) and NfcAdapter#enableReaderMode (requires Android API level 19 or later).
